mailserver/mailserver_db.go

```diff
@@ -0,0 +1,36 @@
+package mailserver
+
+import (
+	whisper "github.com/status-im/whisper/whisperv6"
+	"time"
+)
+
+// dbImpl is an interface to abstract interactions with the db so that the mailserver
+// is agnostic to the underlaying technology used
+type dbImpl interface {
```

Comment:
I think just `mailserver.db` interface name is okay already. Especially since the interface knows nothing about the implementation, idk if we should have `impl` as a part of the name.